### ISSUE TITLE
Generate unique task id's

### ISSUE THREAD

**Issue by egithinji** [tijlleenders/ZinZen-scheduler#75]
The current way task id's are generated is going to be problematic.
1) The 10th task of goal no. 2, and the 1st task of goal no. 21 will have the same task id (210)
2) It involves parsing a string to usize which results in an `unwrap()` (related to #61)

Considering using an `AtomicUsize` (https://doc.rust-lang.org/std/sync/atomic/struct.AtomicUsize.html#) and incrementing it with each generated task.

**Comment by tijlleenders** [comment]
Agree. The task id is just to be able to reference them / keep them apart in the scheduler and the front end.

They will be recalculated every time the scheduler runs and there are no guarantees ids will be consistent when input changes.
